### ISSUE TITLE
vm with multiple pvc disks fails to boot

### ISSUE THREAD

**Issue by karmab** [kubevirt/kubevirt#814]
when launching a vms with multiple disks, qemu fails to boot though the first disk does contain a valid SO on it
the following vm definition

```
apiVersion: kubevirt.io/v1alpha1
kind: VirtualMachine
metadata:
  annotations:
    kcli/plan: kvirt
    kcli/profile: magicdora
    kcli/template: Fedora-Cloud-Base-27-1.6.x86_64.qcow2
    presets.virtualmachines.kubevirt.io/presets-applied: kubevirt.io/v1alpha1
  clusterName: ""
  creationTimestamp: 2018-03-13T14:04:37Z
  generation: 0
  labels:
    kubevirt.io/nodeName: master01.karmalabs.local
  name: mog
  namespace: default
  resourceVersion: "78660"
  selfLink: /apis/kubevirt.io/v1alpha1/namespaces/default/virtualmachines/mog
  uid: 7765c05c-26c7-11e8-ad4f-525400018864
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: sata
        name: disk0
        volumeName: mog-vol0
      - disk:
          bus: sata
        name: disk1
        volumeName: mog-vol1
      - cdrom:
          bus: sata
          readonly: true
          tray: closed
        name: cloudinitdisk
        volumeName: cloudinitvolume
    features:
      acpi:
        enabled: true
    firmware:
      uuid: cc55c8f5-c908-4d50-b440-932373f27785
    machine:
      type: q35
    resources:
      requests:
        memory: 1024M
  terminationGracePeriodSeconds: 0
  volumes:
  - name: mog-vol0
    persistentVolumeClaim:
      claimName: mog-vol0
  - name: mog-vol1
    persistentVolumeClaim:
      claimName: mog-vol1
  - cloudInitNoCloud:
      userDataBase64: I2Nsb3VkLWNvbmZpZwpob3N0bmFtZTogbW9nCnNzaF9wd2F1dGg6IFRydWUKZGlzYWJsZV9yb290OiBmYWxzZQpzc2hfYXV0aG9yaXplZF9rZXlzOgotIHNzaC1yc2EgQUFBQUIzTnphQzF5YzJFQUFBQURBUUFCQUFBQkFRQy9Bdk05VmJPMnlpSWI5QWlsbEJwL2tUcjhqcUlFclJVMUxGS3Fod1BUbTRBdFZJakZTYU91TTRBbHNwZkNVSXo5SUhCckRjWm1iY1lLYWkzbEMzSnRRaWM3TS9hMU9XVWpXRTFNTDhDRXZOc0dQR3U1eU5WVVFvV0MwbG1XNXJ6WDljNkh2SDhBY21mTW1keVE3U2djQW5rMHppcjlqdzhlZDJUUkF6SG4zdlhGZDcrc2FaTGloRkpoWEc0ekI4dmg3Z0pIakxmaklhM0pIcHRXelc5QXRxRjlRc29CWS9pdTU4UmYvaFJucmZXc2N5TjN4OXBHQ1NFcWRMU0R2N0hGdUgyRWFibnZORkZRWnI0SjFGWXpIL2ZLVlkzUHB0M3JmNjRVV0N6dER1N0w0NGZQd3drSTduQXpkbVFWVGFNb0QzRWo4aTcvT1NGWnNDMlY1SUJUIGtib3VtZWRoQGJ1bWJsZWZvb3QKcnVuY21kOgotIGVjaG8gdW5peDEyMzQgfCBwYXNzd2QgLS1zdGRpbiByb290Ci0gZWNobyAiV2VsY29tZSB0byBrdWJldmlydCB3b3JsZCIgPi9ldGMvbW90ZAo=
    name: cloudinitvolume
status:
  interfaces:
  - ipAddress: 10.128.0.21
  nodeName: master01.karmalabs.local
  phase: Running
```
lead to the following dom xml

```
<domain type='qemu' id='1'>
  <name>default_mog</name>
  <uuid>cc55c8f5-c908-4d50-b440-932373f27785</uuid>
  <metadata>
    <kubevirt xmlns="http://kubevirt.io">
      <uid>7765c05c-26c7-11e8-ad4f-525400018864</uid>
      <graceperiod>
        <deletionGracePeriodSeconds>0</deletionGracePeriodSeconds>
      </graceperiod>
    </kubevirt>
  </metadata>
  <memory unit='KiB'>1000448</memory>
  <currentMemory unit='KiB'>1000448</currentMemory>
  <vcpu placement='static'>1</vcpu>
  <resource>
    <partition>/machine</partition>
  </resource>
  <sysinfo type='smbios'>
    <system>
      <entry name='uuid'>cc55c8f5-c908-4d50-b440-932373f27785</entry>
    </system>
  </sysinfo>
  <os>
    <type arch='x86_64' machine='pc-q35-2.10'>hvm</type>
    <boot dev='hd'/>
  </os>
  <features>
    <acpi/>
  </features>
  <clock offset='utc'/>
  <on_poweroff>destroy</on_poweroff>
  <on_reboot>restart</on_reboot>
  <on_crash>destroy</on_crash>
  <devices>
    <emulator>/usr/bin/qemu-system-x86_64</emulator>
    <disk type='file' device='disk'>
      <driver name='qemu' type='raw'/>
      <source file='/var/run/kubevirt-private/vm-disks/mog-vol0/disk.img'/>
      <backingStore/>
      <target dev='sda' bus='sata'/>
      <alias name='sata0-0-0'/>
      <address type='drive' controller='0' bus='0' target='0' unit='0'/>
    </disk>
    <disk type='file' device='disk'>
      <driver name='qemu' type='raw'/>
      <source file='/var/run/kubevirt-private/vm-disks/mog-vol1/disk.img'/>
      <backingStore/>
      <target dev='sdb' bus='sata'/>
      <alias name='sata0-0-1'/>
      <address type='drive' controller='0' bus='0' target='0' unit='1'/>
    </disk>
    <disk type='file' device='cdrom'>
      <driver name='qemu' type='raw'/>
      <source file='/var/run/libvirt/kubevirt-ephemeral-disk/cloud-init-data/default/mog/noCloud.iso'/>
      <backingStore/>
      <target dev='sdc' bus='sata'/>
      <readonly/>
      <alias name='sata0-0-2'/>
      <address type='drive' controller='0' bus='0' target='0' unit='2'/>
    </disk>
    <controller type='usb' index='0' model='nec-xhci'>
      <alias name='usb'/>
      <address type='pci' domain='0x0000' bus='0x03' slot='0x00' function='0x0'/>
    </controller>
    <controller type='sata' index='0'>
      <alias name='ide'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1f' function='0x2'/>
    </controller>
    <controller type='pci' index='0' model='pcie-root'>
      <alias name='pcie.0'/>
    </controller>
    <controller type='pci' index='1' model='dmi-to-pci-bridge'>
      <model name='i82801b11-bridge'/>
      <alias name='pci.1'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1e' function='0x0'/>
    </controller>
    <controller type='pci' index='2' model='pci-bridge'>
      <model name='pci-bridge'/>
      <target chassisNr='2'/>
      <alias name='pci.2'/>
      <address type='pci' domain='0x0000' bus='0x01' slot='0x00' function='0x0'/>
    </controller>
    <controller type='pci' index='3' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='3' port='0x10'/>
      <alias name='pci.3'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x0' multifunction='on'/>
    </controller>
    <controller type='pci' index='4' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='4' port='0x11'/>
      <alias name='pci.4'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x1'/>
    </controller>
    <controller type='pci' index='5' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='5' port='0x12'/>
      <alias name='pci.5'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x2'/>
    </controller>
    <interface type='bridge'>
      <mac address='0a:58:0a:80:00:15'/>
      <source bridge='br1'/>
      <target dev='vnet0'/>
      <model type='e1000'/>
      <alias name='net0'/>
      <address type='pci' domain='0x0000' bus='0x02' slot='0x01' function='0x0'/>
    </interface>
    <serial type='unix'>
      <source mode='bind' path='/var/run/kubevirt-private/default/mog/virt-serial0'/>
      <target port='0'/>
      <alias name='serial0'/>
    </serial>
    <console type='unix'>
      <source mode='bind' path='/var/run/kubevirt-private/default/mog/virt-serial0'/>
      <target type='serial' port='0'/>
      <alias name='serial0'/>
    </console>
    <input type='mouse' bus='ps2'>
      <alias name='input0'/>
    </input>
    <input type='keyboard' bus='ps2'>
      <alias name='input1'/>
    </input>
    <graphics type='vnc' socket='/var/run/kubevirt-private/default/mog/virt-vnc'>
      <listen type='socket' socket='/var/run/kubevirt-private/default/mog/virt-vnc'/>
    </graphics>
    <video>
      <model type='vga' vram='16384' heads='1' primary='yes'/>
      <alias name='video0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x0'/>
    </video>
    <memballoon model='virtio'>
      <alias name='balloon0'/>
      <address type='pci' domain='0x0000' bus='0x04' slot='0x00' function='0x0'/>
    </memballoon>
  </devices>
  <seclabel type='dynamic' model='dac' relabel='yes'>
    <label>+107:+107</label>
    <imagelabel>+107:+107</imagelabel>
  </seclabel>
</domain>
```

as a comparison, on a local domxml from  a standard kvm host, the disks are defined with 

*\<address type='pci' domain='0x0000' bus='0x00' slot='0x04' function='0x0'/\>*

**Comment by rmohr** [comment]
@karmab If you still have the environment at hand, could you run `kubectl describe mog`, so that we can see why qemu/libvirt complains in the events?

**Comment by rmohr** [comment]
BTW. from what you posted it looks like qemu is running, so does `qemu` or the `vm` fail to boot?

**Comment by karmab** [comment]
qemu is running, vm also in state running, it s just that OS doesnt boot

**Comment by karmab** [comment]
i dont have the vm but can recreate the issue easily, but there s nothing relevant in the describe of the vm, since from a k8s point of view, everything is OK

**Comment by rmohr** [comment]
Yep, then I don't need the `describe` output. Are the images in the PVC `raw` images?

**Comment by karmab** [comment]
yes, both images are raw:
- one is created with a pod that does a copy between pvcs, where the origin pvc got created with a disk-importer pod.
- the other is created with a pod running fallocate to put a disk.img with the desired size

**Comment by rmohr** [comment]
@petrkotas I suspect that our controller messes something up when we define deh PVC mount points on our launcher pod. Could you have a look and add a test which ensures that adding two PVCs works as expected?

**Comment by fabiand** [comment]
Does this only affetc multiple PVC disks, or also PVC and CloudInit

**Comment by karmab** [comment]
in my tests, i always used cloudinit
so:
- pvc+cloudinit works 
- pvc+pvc+cloudinit doesnt
note i present cloudinit as a cdrom though

**Comment by petrkotas** [comment]
Will look at this. ASAP. I am finishing other task in oVirt.

--
s pozdravem,
Petr Kotas

**Comment by petrkotas** [comment]
Test to verify the functionality is introduced in #843.
The test is passing without any complication. 

However I was able to reproduce the issue described above two times.
First: right from start the setup behaved exactly as described above. I was able to solve the issue by destroying the test setup and creating it again. This case was caused by inconsistent deployment of kubevirt.

Second: the issue occurred once more after roughly 3 hours of testing. This time I was able to just restart vagrant and it worked as expected. It seems that the vagrant networking is not very stable and at some point it stop working.

I am proposing to close this issue, @rmohr @karmab 

I propose closing this issue.

**Comment by rmohr** [comment]
> I propose closing this issue.

Could be that you were then hitting #837. Let's keep this open until @karmab can confirm that it works for him too.

**Comment by petrkotas** [comment]
@rmohr ack

**Comment by nellyc** [comment]
@fabiand any updates?

**Comment by petrkotas** [comment]
This issue has already been solved by the #843.
I would propose to close this issue since @karmab seems to have it all
running.
--
s pozdravem,
Petr Kotas

**Comment by fabiand** [comment]
Thanks, closing this!
